src/app/clusters/wifi-network-diagnostics-server/wifi-network-diagnostics-server.cpp

```diff
@@ -240,6 +247,25 @@ CHIP_ERROR WiFiDiagosticsAttrAccess::Read(const ConcreteReadAttributePath & aPat
     return CHIP_NO_ERROR;
 }
 
+void WiFiDiagosticsGlobalInstance::InvokeCommand(HandlerContext & handlerContext)
+{
+    switch (handlerContext.mRequestPath.mCommandId)
+    {
+    case Commands::ResetCounts::Id:
+        CommandHandlerInterface::HandleCommand<Commands::ResetCounts::DecodableType>(
+            handlerContext, [this](HandlerContext & ctx, const auto & commandData) { HandleResetCounts(ctx, commandData); });
+        break;
+    }
+}
+
+void WiFiDiagosticsGlobalInstance::HandleResetCounts(HandlerContext & ctx, const Commands::ResetCounts::DecodableType & commandData)
+{
+    DeviceLayer::GetDiagnosticDataProvider().ResetWiFiNetworkDiagnosticsCounts();
+    ctx.mCommandHandler.AddStatus(ctx.mRequestPath, Protocols::InteractionModel::Status::Success);
+}
+
+WiFiDiagosticsGlobalInstance gWiFiDiagosticsInstance;
```

Comment:
If this remains a global it should be `constexpr` or use `chip::Global` (see Global.h).

Reply:
constexpr does not work for WiFiDiagosticsGlobalInstance since there is runtime dependencies that invalidate compile-time initialization.

chip::Global<T> is typically used to defer or control construction at runtime to avoid global initialization ordering issues .

WiFiDiagosticsGlobalInstance only initialized with DeviceLayer::GetDiagnosticDataProvider(), can be constructed entirely at compile time